### PR TITLE
Rewrite configuration / environment logic using `@wuespace/envar`

### LINKED ISSUE #4: Migrate configuration to `@wuespace/envar`
Since we now released https://jsr.io/@wuespace/envar, we should migrate the generator to also use it for its configuration.

### LINKED ISSUE #8: `zBoolString()` breaks apart when parsing already `boolean` values.
That, in and of itself, wouldn't really be an issue. However, given that we also parse the unserialized config from the database using that very same schema, that becomes ... a bit of an issue. So yeah: it's an issue.

https://github.com/wuespace/deno-asn-generator/blob/fc6c2875578bf53952e063aec02b59c71b1d504c/lib/common/config.ts#L236

### PR DESCRIPTION
**Pull request by pklaschka**
Fixes all kinds of glaring problems the previous configuration had, including:

1. problems related to using the same schema for env and DB config parsing,
2. bad testability due to globally static `CONFIG` object, and
3. bad documentation of configuration variables due to complex structure.

With this, we can also use `@wuespace/envardoc` to document our env variables.

- Closes #4
- Closes #8

- `main`
  - \#10 :point\_left:

### CODE REVIEW

File: lib/http/routes/lookup.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ lookupRoutes.post(
     return parsed.data;
   }),
   (c) => {
-    const asn = CONFIG.ASN_PREFIX + c.req.valid("form").asn;
+    const asn = getConfig().ASN_PREFIX + c.req.valid("form").asn;
```
[the remark this answers is not in the document]
Comment (pklaschka) [reply lib/http/routes/lookup.ts:25]: The config is only used once in this function.

File: lib/http/routes/lookup.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ lookupRoutes.get(
   (c) => {
     const asn = c.req.valid("param").asn;
 
-    if (!CONFIG.ASN_LOOKUP_URL) {
+    if (!getConfig().ASN_LOOKUP_URL) {
```
[the remark this answers is not in the document]
Comment (pklaschka) [reply lib/http/routes/lookup.ts:41]: The config is only used once in that function.

File: lib/http/lookup-url.ts [hunk 3]
```diff
@@ -17,8 +17,8 @@ export function getLookupURL(asn: string): string {
     throw new Error("Invalid ASN");
   }
 
-  if (!CONFIG.ASN_LOOKUP_INCLUDE_PREFIX) {
-    asn = asn.slice(CONFIG.ASN_PREFIX.length);
+  if (!config.ASN_LOOKUP_INCLUDE_PREFIX) {
+    asn = asn.slice(config.ASN_PREFIX.length);
```
[the remark this answers is not in the document]
Comment (pklaschka) [reply lib/http/lookup-url.ts:21]: Validated by configuration:

https://github.com/wuespace/deno-asn-generator/blob/13b890062d9237b52350286eebb996492fd1b65f/lib/common/config.ts#L169

https://github.com/wuespace/deno-asn-generator/blob/13b890062d9237b52350286eebb996492fd1b65f/lib/common/config.ts#L146

**Review by fussel178 — APPROVED** [review]
LGTM :hut:

**Pull request merged**